http://bugs.gentoo.org/132763
http://savannah.gnu.org/bugs/?func=detailitem&item_id=14407
http://savannah.gnu.org/bugs/?func=detailitem&item_id=14758

2005-11-28  Tim Deegan <Tim.Deegan@cl.cam.ac.uk>

	* stage2/disk_io.c: Fixed a bug in partition table 
	offset tracking (#14407/#14758).

diff -Nru grub-20051128/stage2/disk_io.c grub-ptable-fix/stage2/disk_io.c
--- grub-20051128/stage2/disk_io.c	2004-05-23 17:45:45.000000000 +0100
+++ grub-ptable-fix/stage2/disk_io.c	2005-11-28 13:10:28.659511000 +0000
@@ -800,7 +800,12 @@
 
   /* If this is the whole disk, return here.  */
   if (! flags && current_partition == 0xFFFFFF)
-    return 1;
+    {
+#ifndef STAGE1_5
+      cur_part_offset = 0;
+#endif /* ! STAGE1_5 */
+      return 1;
+    }
 
   if (flags)
     dest_partition = 0xFFFFFF;
